app/assets/javascripts/discourse-common/addon/utils/dom-utils.js

```diff
@@ -0,0 +1,51 @@
+function isWindow(obj) {
+  return obj != null && obj === window;
+}
+
+function scrollTop(element) {
+  return element
+    ? element.scrollTop
+    : window.pageYOffset || document.documentElement.scrollTop;
```

Comment:
In what cases does `|| document.documentElement.scrollTop` kick in?

Reply:
yes it shouldn't be needed in modern browsers, Im kinda careful there, but I'm ok to just go with pageYOffset I also can't find a way to need the other

Reply:
Alternatively, we could avoid using `window` and rely on `documentElement` instead. 🙂 (then we won't need that helper either 😛)